operator/parser/csv/csv_test.go

```diff
@@ -572,29 +572,176 @@ func TestParserCSV(t *testing.T) {
 	}
 }
 
-func TestParserCSVMultipleBodies(t *testing.T) {
-	t.Run("basic", func(t *testing.T) {
-		cfg := NewCSVParserConfig("test")
-		cfg.OutputIDs = []string{"fake"}
-		cfg.Header = testHeader
+func TestParserCSVMultiline(t *testing.T) {
+	cases := []struct {
+		name     string
+		input    string
+		expected map[string]interface{}
+	}{
+		{
+			"first_field",
+			"aa\naa,bbbb,cccc,dddd,eeee",
+			map[string]interface{}{
+				"A": "aa\naa",
+				"B": "bbbb",
+				"C": "cccc",
+				"D": "dddd",
+				"E": "eeee",
+			},
+		},
+		{
+			"middle_field",
+			"aaaa,bbbb,cc\ncc,dddd,eeee",
+			map[string]interface{}{
+				"A": "aaaa",
+				"B": "bbbb",
+				"C": "cc\ncc",
+				"D": "dddd",
+				"E": "eeee",
+			},
+		},
+		{
+			"last_field",
+			"aaaa,bbbb,cccc,dddd,e\neee",
+			map[string]interface{}{
+				"A": "aaaa",
+				"B": "bbbb",
+				"C": "cccc",
+				"D": "dddd",
+				"E": "e\neee",
+			},
+		},
+		{
+			"multiple_fields",
+			"aaaa,bb\nbb,ccc\nc,dddd,e\neee",
+			map[string]interface{}{
+				"A": "aaaa",
+				"B": "bb\nbb",
+				"C": "ccc\nc",
+				"D": "dddd",
+				"E": "e\neee",
+			},
+		},
+		{
+			"multiple_first_field",
+			"a\na\na\na,bbbb,cccc,dddd,eeee",
+			map[string]interface{}{
+				"A": "a\na\na\na",
+				"B": "bbbb",
+				"C": "cccc",
+				"D": "dddd",
+				"E": "eeee",
+			},
+		},
+		{
+			"multiple_middle_field",
+			"aaaa,bbbb,c\nc\nc\nc,dddd,eeee",
+			map[string]interface{}{
+				"A": "aaaa",
+				"B": "bbbb",
+				"C": "c\nc\nc\nc",
+				"D": "dddd",
+				"E": "eeee",
+			},
+		},
+		{
+			"multiple_last_field",
+			"aaaa,bbbb,cccc,dddd,e\ne\ne\ne",
+			map[string]interface{}{
+				"A": "aaaa",
+				"B": "bbbb",
+				"C": "cccc",
+				"D": "dddd",
+				"E": "e\ne\ne\ne",
+			},
+		},
+		{
+			"leading_newline",
+			"\naaaa,bbbb,cccc,dddd,eeee",
+			map[string]interface{}{
+				"A": "aaaa",
+				"B": "bbbb",
+				"C": "cccc",
+				"D": "dddd",
+				"E": "eeee",
+			},
+		},
+		{
+			"trailing_newline",
+			"aaaa,bbbb,cccc,dddd,eeee\n",
+			map[string]interface{}{
+				"A": "aaaa",
+				"B": "bbbb",
+				"C": "cccc",
+				"D": "dddd",
+				"E": "eeee",
+			},
+		},
+		{
+			"leading_newline_field",
+			"aaaa,\nbbbb,\ncccc,\ndddd,eeee",
+			map[string]interface{}{
+				"A": "aaaa",
+				"B": "\nbbbb",
+				"C": "\ncccc",
+				"D": "\ndddd",
+				"E": "eeee",
+			},
+		},
+		{
+			"trailing_newline_field",
+			"aaaa,bbbb\n,cccc\n,dddd\n,eeee",
+			map[string]interface{}{
+				"A": "aaaa",
+				"B": "bbbb\n",
+				"C": "cccc\n",
+				"D": "dddd\n",
+				"E": "eeee",
+			},
+		},
+		{
+			"empty_lines",
+			"aa\n\naa,bbbb,c\n\nccc,dddd,eee\n\ne",
+			map[string]interface{}{
+				"A": "aa\naa",
+				"B": "bbbb",
+				"C": "c\nccc",
+				"D": "dddd",
+				"E": "eee\ne",
+			},
+		},
+		{
+			"everything",
+			"\n\na\na\n\naa,\n\nbb\nbb\n\n,cc\ncc\n\n,\ndddd\n,eeee\n\n",
+			map[string]interface{}{
+				"A": "a\na\naa",
+				"B": "\nbb\nbb\n",
```

Comment:
shouldn't this value end with two `\n`?

Reply:
Actually, not in this case, but I added a case where the empty line is preserved. This is actually covered in [RFC 4180](https://datatracker.ietf.org/doc/html/rfc4180) `Fields containing line breaks (CRLF), double quotes, and commas should be enclosed in double-quotes`

Reply:
makes sense